### PR TITLE
Builtins

### PR DESCRIPTION
**Pull request by Naruto0**
Attemp to finish builtins stubs.

### CODE REVIEW

File: stdlib/2.7/__builtin__.pyi [hunk 1]
```diff
@@ -883,7 +883,7 @@ class file(BinaryIO):
 @overload
 def apply(object: object, args: Iterable) -> Any: ...
 @overload
-def apply(object: object, args: Iterable, kwargs: Dict[str, Any]) -> Any: ...
+def apply(object: object, *args: Iterable, **kwargs: Dict[str, Any]) -> Any: ...
```
Comment (Naruto0) [inline stdlib/2.7/__builtin__.pyi]: Maybe i can get rid of overload if *args and **kwargs are arbitrary...

File: stdlib/2.7/__builtin__.pyi [hunk 2]
```diff
@@ -11,6 +11,7 @@ from typing import (
     MutableSet
 )
 from abc import abstractmethod, ABCMeta
+from numbers import Number as _Number
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: In a stub you can just say `from numbers import Number` and it won't be considered exported. In fact, using the `as` form causes the checker to export `_Number`. (See PEP 484, section on Stub Files, under Additional Notes.)

File: stdlib/2.7/__builtin__.pyi [hunk 3]
```diff
@@ -879,3 +880,15 @@ class file(BinaryIO):
     def write(self, data: str) -> None: ...
     def writelines(self, data: Iterable[str]) -> None: ...
     def truncate(self, pos: int = ...) -> int: ...
+
+def apply(object: object, *args: Iterable, **kwargs: Dict[str, Any]) -> Any: ...
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: Actually this should probably be something line
```
def apply(func: Callable[[...], _T], args: Sequence[Any] = None, kwds: Mapping[str, Any] = None) -> _T: ...
```
I also strongly advise you to write some test code that uses apply() and play with it in CPython so you know what it does, and then double-check thtat that code is accepted by mypy with these stubs.

File: stdlib/2.7/__builtin__.pyi [hunk 4]
```diff
@@ -879,3 +880,15 @@ class file(BinaryIO):
...
+
+def coerce(x: _Number, y: _Number) -> Tuple[_Number,_Number]: ...
+
+def help(str) -> None: ...
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: Can you give the argument a type (should be Any or object) and a different name than `str`?

File: stdlib/2.7/__builtin__.pyi [hunk 5]
```diff
@@ -879,3 +880,15 @@ class file(BinaryIO):
     def write(self, data: str) -> None: ...
     def writelines(self, data: Iterable[str]) -> None: ...
     def truncate(self, pos: int = ...) -> int: ...
+
+def apply(object: object, *args: Iterable, **kwargs: Dict[str, Any]) -> Any: ...
+
+def coerce(x: _Number, y: _Number) -> Tuple[_Number,_Number]: ...
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: Sadly this doesn't work -- mypy doesn't appear to consider the concrete numeric types (int, float etc.) to be instances of Number.

Maybe a type variable would work?
```
_N = TypeVar('_N', int, float, complex)
def coerce(x: _N, y: _N) -> Tuple[_N, _N]: ...
```
???

**Comment by gvanrossum** [comment]
Ping? Are you going to update this or should we just give up?

**Comment by Naruto0** [comment]
Im working on it.

**Comment by gvanrossum** [comment]
Sorry, I'm giving up on this PR. I've added apply() and coerce() myself (based on my own comments above) in 5c881a6b865eb60a2243d910817393cb7eb218c7. Regarding help(), copyrights(), credits(), license(), those are meant for interactive use only, and in fact when used in a script they are really annoying, so I think it's better not to add them.

**Comment by Naruto0** [comment]
No, I'm sorry. I had problem to finish this up, so it's OK with me. Thank you, Guido.

**Pull request closed without merging**